src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionContextExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace NServiceBus.Routing.MessageDrivenSubscriptions
+{
+    using NServiceBus.Routing;
+
+    /// <summary>
+    /// Context extensions for message driven pubsub
+    /// </summary>
+    public static class MessageDrivenSubscriptionContextExtensions
+    {
+        /// <summary>
+        /// Allows the user to specify a explicit address when subscribing
+        /// </summary>
+        public static void UseExplicitPublisherAddress(this SubscribeContext context,string address)
```

Comment:
@danielmarbach I've been rethinking this, since this wasn't possible before and no one have asked for it I'll call YAGNI and remove it. We can always add it later if needed